### ISSUE TITLE
Plugin stuck in figma

### ISSUE THREAD

**Issue by ZeroCool00** [react-figma/react-figma-boilerplate#5]
I just clone this repo, and load the plugin in figma, but its not rendering anything, it stuck at ```Running react-figma-bilerplate``` toast message

![Simulator Screen Shot - iPhone 11 Pro - 2019-10-06 at 12 48 34](https://user-images.githubusercontent.com/18139594/66268481-607d4080-e85b-11e9-9272-649d8722622c.png)

**Comment by ilyalesik** [comment]
Hi, @ZeroCool00! Can you open console (Plugins -> Development -> Open Console), launch plugin and make a screenshot of console? I expect some errors.

**Comment by ZeroCool00** [comment]
I think is rendering successfully but not getting focus directly to that page, and you forget to close loader afterward, or maybe it skip some condition.

<img width="1397" alt="Screenshot 2019-10-07 at 10 33 13 AM" src="https://user-images.githubusercontent.com/18139594/66286269-3d4fa100-e8ee-11e9-8255-92485014fc49.png">

**Comment by ZeroCool00** [comment]
As you can see i get the result but its not getting focus, and loader keep running.

<img width="923" alt="Screenshot 2019-10-07 at 10 38 00 AM" src="https://user-images.githubusercontent.com/18139594/66286339-9ddede00-e8ee-11e9-90a8-22735e675f04.png">

**Comment by ZeroCool00** [comment]
How about this line, i cannot find anywhere
``` figma.currentPage.selection = nodes;
    figma.viewport.scrollAndZoomIntoView(nodes);
    figma.closePlugin();```

**Comment by ilyalesik** [comment]
The problem really has a place to be. I created a new issue for infinite loading behavior. 

But, I'm afraid it's pretty hard to fix because we don't know, when exactly plugin should stop. As an example we can have code like this:

```javascript
const Component = () => {
   const [someFlag, setSomeFlag] = React.useState(false);
   React.useEffect(() => {
     const timeoutId = setTimeout(() => {
        setSomeFlag(true);
     }, 5000);
     return () => clearTimeout(timeoutId);
   }, []);
   return <Rectangle style={{backgroundColor: someFlag ? green : red}} />
}
```
if we just call `figma.closePlugin();` after `render`, rectangle color never set to `green`.

**Comment by ZeroCool00** [comment]
you can do something like this

```
const Component = () => {
   const [someFlag, setSomeFlag] = React.useState(false);
   React.useEffect(() => {
     const timeoutId = setTimeout(() => {
        setSomeFlag(true);
     }, 5000);
     return () => clearTimeout(timeoutId) | parent.postMessage({ pluginMessage: 'close' }, '*'); /// **Pass this when time over**
  }, []);
   return <Rectangle style={{backgroundColor: someFlag ? green : red}} />
}
```

And in onMessage you can check

```
figma.ui.onmessage = message => {
  const nodes = [];
  subscribeOnMessages(message);
  if (message === 'close') {
     figma.closePlugin();
 }
};
```

**Comment by ZeroCool00** [comment]
@ilyalesik  What about  `New Page not getting focus` issue.

**Comment by ilyalesik** [comment]
> What about New Page not getting focus issue

I agree we should have some way to make page current. But, I think, we can't just make every new page current for many reasons. As an example, react-figma allows several page creation:

```javascript
const App = () => {
    return <>
        <Page name="Page 1" />
        <Page name="Page 2" />
   </>
}
```

Explicit API needed, e.g. `isCurrent` prop for `Page`.
I created [#63](https://github.com/react-figma/react-figma/issues/63).

> parent.postMessage({ pluginMessage: 'close' }, '*');  // *Here*

It's an interesting idea! But we can have several components with async actions. After the first completed effect plugin will close and the remaining effects will not complete. It's hard to control because `react-figma` has internal react hooks with effects (e.g. `useFillsPreprocessor`, `useYogaLayout`) that used inside components.
